react/SelectBox/SelectBox.jsx

```diff
@@ -216,22 +216,57 @@ ActionsOption.defaultProps = {
   actions: []
 }
 
+// With React, the referenced element is in the current property.
+// With Preact, the referenced element is the object
+const getNodeFromRef = ref => {
+  return (ref && ref.current) || ref
+}
+
+const customComponents = {
+  DropdownIndicator,
+  Option
+}
+
 class SelectBox extends Component {
-  state = { isOpen: false }
-  element = null
+  constructor(props) {
+    super(props)
+    this.state = { isOpen: false }
+    this.element = null
+    this.handleRef = this.handleRef.bind(this)
+    this.handleOpen = this.handleOpen.bind(this)
+    this.handleClose = this.handleClose.bind(this)
+  }
+
+  handleOpen() {
+    this.refreshMenuStyle()
+    this.setState({
+      isOpen: true
+    })
+  }
 
-  handleOpen = () => {
-    this.setState({ isOpen: true })
+  refreshMenuStyle() {
+    if (this.props.container && this.controlRef) {
+      this.menuStyle = computedMenuListHeightStyles(
+        getNodeFromRef(this.props.container),
+        this.controlRef
+      )
+    }
   }
 
-  handleClose = () => {
```

Comment:
This way to write is not handled in cozy-ui?

Reply:
It is but it is harder to write tests and the lead dev of enzyme gives advice against it. More info on https://github.com/cozy/cozy-guidelines/pull/58